lib/level10/games/games.ex

```diff
@@ -4,110 +4,120 @@ defmodule Level10.Games do
   end
 
-  @spec put_new_hands(Game.t()) :: Game.t()
-  defp put_new_hands(%{draw_pile: deck, players: players} = game) do
-    {hands, draw_pile} =
-      Enum.reduce(players, {%{}, deck}, fn %{name: player_name}, {hands, draw_pile} ->
-        {player_hand, rest_of_pile} = Enum.split(draw_pile, 10)
-        hands = Map.put(hands, player_name, player_hand)
-        {hands, rest_of_pile}
-      end)
+  @spec do_create_game(Player.t(), non_neg_integer()) ::
+          {:ok, Game.join_code(), Player.id()} | :error
+  defp do_create_game(player, attempts_remaining)
 
-    %{game | draw_pile: draw_pile, hands: hands}
+  defp do_create_game(_player, 0) do
+    :error
   end
 
-  @spec new_deck() :: Card.t()
-  defp new_deck do
-    color_cards =
-      for value <- ~W[one two three four five six seven eight nine ten eleven twelve wild]a,
-          color <- ~W[blue green red yellow]a,
-          card = Card.new(value, color),
-          _repeat <- 1..2 do
-        card
-      end
+  defp do_create_game(player, attempts_remaining) do
+    join_code = Game.generate_join_code()
 
-    skips = for _repeat <- 1..4, do: Card.new(:skip, :blue)
+    case Agent.start_link(Game, :new, [join_code, player], name: via(join_code)) do
+      {:ok, _pid} ->
+        {:ok, join_code, player.id}
 
-    color_cards
-    |> Stream.concat(skips)
-    |> Enum.shuffle()
+      {:error, {:already_started, _pid}} ->
+        do_create_game(player, attempts_remaining - 1)
+    end
   end
 
-  @doc """
-  At the end of a round, the game struct should be passed into this function.
-  It will update player scoring and levels, check if the game has been
-  complete, and reset the state for the next round.
-  """
-  @spec complete_round(Game.t()) :: Game.t()
-  def complete_round(game) do
-    game
-    |> update_scoring_and_levels()
-    |> check_complete()
-    |> clear_round()
+  @spec via(Game.join_code()) :: game_name()
+  defp via(join_code) do
+    {:via, Registry, {GameRegistry, join_code}}
   end
 
-  @spec update_scoring_and_levels(Game.t()) :: Game.t()
-  defp update_scoring_and_levels(%{scoring: scoring, table: table, hands: hands} = game) do
-    new_scoring =
-      scoring
-      |> Enum.map(fn {player, {level, score}} ->
-        player_hand = hands[player]
-        score_for_hand = player_hand |> Enum.map(&Card.score/1) |> Enum.sum()
-        new_score = score + score_for_hand
+  @spec join_game(Game.join_code(), String.t()) :: {:ok, Player.id()} | :already_started
+  def join_game(join_code, player_name) do
+    player = Player.new(player_name)
 
-        player_advanced = !is_nil(table[player])
-        new_level = if player_advanced, do: level + 1, else: level
+    Agent.get_and_update(via(join_code), fn game ->
+      case Game.put_player(game, player) do
+        {:ok, game} ->
+          {{:ok, player.id}, game}
 
-        {player, {new_level, new_score}}
-      end)
-      |> Enum.into(%{})
-
-    %{game | scoring: new_scoring}
+        :already_started ->
```

Comment:
We'll probably want to handle rejoining here, but we can worry about that later...